File: LeetCode/Weekly Contest 390/solution_3.cs
public class Solution {
    public long[] MostFrequentIDs(int[] nums, int[] freq) {
        int n = nums.Length;
        long[] ans = new long[n];
        
        Dictionary<int, int> dict = new();
        
        for(int i = 0; i < n; i++){
            
            if(dict.ContainsKey(nums[i])){
                dict[nums[i]] += freq[i];
            }else{
                dict.Add(nums[i], freq[i]);
            }
            
            int maxFreq = 0;
            foreach(var pair in dict){
                if(pair.Value > maxFreq)
                    maxFreq = pair.Value;
            }
            
            ans[i] = maxFreq;
            
        }
        
        return ans;
    }
}

File: LeetCode/Biweekly Contest 133/solution3.cpp
#include <vector>
using namespace std;


class Solution {
public:
    int minOperations(vector<int>& nums) {
        int n = nums.size();        
        int numOfFlipping = 0;

        for(int i = 0; i < n; i++){
            if(nums[i] == 0 && (numOfFlipping % 2 == 0) ||
               nums[i] == 1 && (numOfFlipping % 2 == 1)){
                numOfFlipping++;
            }
        }

        return numOfFlipping;
    }
};